### PR TITLE
Add offboarding checklists

### PR DESCRIPTION
**Pull request by mercadon**

### CODE REVIEW

**Comment by nicksnyder** [comment]
I am worried about these offboarding checklists getting out of date and desynchronized with the onboarding checklists.

Is there anything on the offboarding checklist that doesn't map to an item in an onboarding checklist?

If not, I propose that we just maintain an onboarding checklist and the offboarding process is to execute the onboarding checklist except take the opposite action.

This would be easier if we rephrased the list items. For example: "on call rotation" not "add to oncall rotation", "recurring 1-1" not "create a recurring 1-1".

Then these list items can be copied into an issue/doc titled onboarding or offboarding and it is obvious what action to do.

**Comment by mercadon** [comment]
> Is there anything on the offboarding checklist that doesn't map to an item in an onboarding checklist?
There are ~3 things that don't map to the onboarding checklist, and there are ~8 things from that onboarding list that don't map to the offboarding checklist. I hear you though– it is an extra list to keep updated and will require cross-referencing these two checklists, but I think there are enough differences that if we didn't have separate docs some items/actions would be unclear. 

For now, I think it'll be valuable to have both an onboarding and offboarding checklist– if it does become too much of a burden to keep them both updated I'm happy to revisit combining them, so long as it's clear what belongs to onboarding only or onboarding+offboarding.

File: handbook/people-ops/offboarding/engineers.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- Revoke access to necessary services.
+  - [Sourcegraph organization on GitHub](https://github.com/orgs/sourcegraph/people)
+    - Inlcuding any relevant GitHub teams, including [@sourcegraph/everyone](https://github.com/orgs/sourcegraph/teams/everyone).
+  - [LSIF organization on GitHub](https://github.com/orgs/lsif/people)
+  - [Buildkite](https://buildkite.com/organizations/sourcegraph/users/new)
+  - Google Cloud Platform ([prod](https://console.cloud.google.com/iam-admin/iam?project=sourcegraph-dev), [test](https://console.cloud.google.com/iam-admin/iam?project=sourcegraph-server))
+  - [Opsgenie](https://sourcegraph.app.opsgenie.com/settings/users/)
+  - [Docker Hub](https://hub.docker.com/orgs/sourcegraph)
+- Remove recurring 1-1.
+- Schedule time to discuss Sourcegraph's system architecture and tech stack.
```
Comment (nicksnyder) [inline handbook/people-ops/offboarding/engineers.md:14]:
```suggestion
- Remove site admin privileges from Sourcegraph.com
```

**Comment by sqs** [comment]
@mercadon Do you still intend to merge this PR? If not, please close it. I'll close it in a couple days if I haven't heard back (I'm just trying to clean up our PR list).

**Pull request closed without merging**